File: type-script-nest-graph-ql-실습/src/app/role/model/dto/response/role-response.dto.ts
import { RoleEntity } from "../../entities/role.entity";

export class RoleResponseDto {
  private roleId: string;
  private job: string;
  private requirement: string;

  constructor(entity: RoleEntity) {
    this.roleId = entity.roleId;
    this.job = entity.job;
    this.requirement = entity.requirement;
  }
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/role/model/dto/response/RoleResponseDto.kt
package com.junyharangstudy.kotlingraphqltest.api.role.model.dto.response

import com.junyharangstudy.kotlingraphqltest.api.role.model.entity.Role

class RoleResponseDto (
    private val roleId: String,
    private val job: String,
    private val requirement: String
) {
    companion object {
        fun roleToDto(role: Role): RoleResponseDto {
            return RoleResponseDto(
                roleId = role.roleId,
                job = role.job,
                requirement = role.requirement)
        }
    }
}